---
title: "Evaluación de la selección genómica de un solo paso en arroz"
output:
  xaringan::moon_reader:
    css: [default, css/cc-fonts.css, css/Cajitas.css]
    lib_dir: libs
    seal: false
    nature:
      #titleSlideClass: ["center", "top", "my-title"]
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      slideNumberFormat: "%current%<br>"
---
class: center

```{r Paquetes usados, general y bibliografia, load_refs, eval = TRUE, echo = FALSE, include = FALSE, cache = FALSE}

# Paquetes usados ----

pacman::p_load(metathis, xaringanExtra, fontawesome, RefManageR, here, tidyverse, latex2exp, showtext, ggforce, magick, pedigreemm, reshape2, gt) # Aquí (https://pkg.garrickadenbuie.com/xaringanExtra/#/) y aquí (https://github.com/gadenbuie/xaringanExtra) explican las muchas opciones de usar el paquete xaringanExtra.
font_add_google('Gochi Hand', 'gochi')
showtext_auto()

# Opciones generales de la presentación ----

workshop_day <- format(as.Date("2022-06-01"), format = "%B/%d/%Y")

# Habilita la capacidad de mostrar todas las diapositivas en una vista general de mosaico presionando "o"
xaringanExtra::use_tile_view()
xaringanExtra::use_panelset()
xaringanExtra::use_clipboard()
xaringanExtra::use_share_again()
xaringanExtra::style_share_again(share_buttons = "all")
xaringanExtra::use_extra_styles(
  hover_code_line = TRUE,
  mute_unhighlighted_code = FALSE
)

# Opciones de bibliografía ----

options(htmltools.dir.version = FALSE)

BibOptions(check.entries = FALSE, bib.style = "authoryear", style = "markdown",
           dashed = TRUE)

mi_Bib <- ReadBib(here('Presentacion', 'Bibliografia', 'mi_Bib.bib'), check = FALSE)

# Característica ¡banner! ----

#xaringanExtra::use_banner(
  #top_left = "Evaluación de la selección genómica de un solo paso en arroz",
  #top_right = "Leo",
  #bottom_left = "leo4luffy.github.io/Repo_TFM/",
  #exclude = "title-slide"
#)
```

```{r metathis, eval = TRUE, echo = FALSE}

meta() %>%
  meta_general(
    description = "Presentación TFM",
    generator = "xaringan and remark.js"
  ) %>%
  meta_name('github-repo' = 'Leo4Luffy/Presentacion_TFM') %>% 
  meta_social(
    title = "Presentación del trabajo de fin de máster",
    url = "https://leo-tfm.netlify.app/",
    og_type = "website",
    og_author = "Jorge Leonardo López Martínez",
    twitter_card_type = "summary_large_image",
    twitter_creator = "@leo4luffy"
  )
```

```{r xaringanExtra-share-again, eval = FALSE, echo = FALSE}

xaringanExtra::use_share_again()
```

###.fancy[`r rmarkdown::metadata$title`] 

<!--###.fancy[`r rmarkdown::metadata$subtitle`]-->

<br>
<br>

<P align="justify"> **Estudiante:** Jorge Leonardo López Martínez <p>
<!--<P align="justify">[`r fa(name = "github", fill = "black")` Leo4Luffy](https://github.com/Leo4Luffy)<p>-->

<br>

<P align="justify">**Director:** Miguel Pérez-Enciso<p>
<!--<P align="justify">[`r fa(name = "github", fill = "black")` miguelperezenciso](https://github.com/miguelperezenciso)<p>-->


<br>
<br>
<br>

<!--<P align="justify">`r workshop_day`<p>-->

```{r, echo = FALSE, out.width = "80%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'Portada.png'))
```

---
class: center

### Introducción

.panelset.sideways[

.panel[.panel-name[Mejor predictor lineal insesgado]

<P align="justify">El BLUP es un método que se utiliza para predecir el valor genético de un individuo. Para ello, se usa una matriz de parentesco estimada a partir de la información del pedigrí<sup>1</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

data.frame(
  x = 4.5, y = 0.5
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 4.25, ymin = 3.75, ymax = 5.75, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 2.875, y = 4.75, label = 'A', size = 9.4, colour = 'black') +
  annotate(geom = 'text', x = 1.1, y = 4.75, label = 'Matriz A', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 4.1, xmax = 6.3, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 5.2, y = 1.8, label = 'BLUP', size = 7.6, colour = 'black') +
  
  geom_curve(x = 6.4, xend = 7.5, y = 1.8, yend = 1.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  annotate(geom = 'text', x = 8.4, y = 1.8, label = 'EBV', size = 7.6, colour = 'black') +
  
  geom_curve(x = 2.875, xend = 3.95, y = 3.45, yend = 2.1, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  scale_x_continuous(limits = c(0.45, 9.4)) +
  scale_y_continuous(limits = c(1.0, 6.3)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 11, face = 'bold'))
```

]

.panel[.panel-name[Matriz de parentesco- I]

<P align="justify">En la matriz **A**, los elementos fuera de la diagonal indican la relación entre individuos<sup>2</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

Puntos <- data.frame(x = c(0.65, 0.25, 1.75, 0.75, 1.75), y = c(1.75, 1.4, 1.4, 1.05, 1.05))
Cuadros <- data.frame(x = c(1.35, 0.75, 1.25, 0.25, 1.25), y = c(1.75, 1.4, 1.4, 1.05, 1.05))

ggplot() +
  
  annotate(geom = 'rect', xmin = 0.0, xmax = 2.0, ymin = 1.9, ymax = 1.2, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 3.35, xmax = 3.85, ymin = 0.1, ymax = 0.3, colour = 'yellow', fill = 'yellow', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 3.95, xmax = 4.45, ymin = -0.1, ymax = 0.1, colour = 'yellow', fill = 'yellow', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  geom_point(aes(x = x, y = y), data = Puntos, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadros, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.8, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.8, label = '0.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.6, label = '0.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.6, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.4, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.2, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.0, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = -0.2, label = '1.00', size = 6.8, colour = 'black') +
  
  scale_x_continuous(limits = c(0.0, 4.45)) +
  scale_y_continuous(limits = c(-0.3, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

.panel[.panel-name[Matriz de parentesco- II]

<P align="justify">El parentesco que proviene al usar la matriz **A** es un parentesco esperado, lo cual no refleja el porcentaje real de genes idénticos entre individuos emparentados<sup>1</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

Puntos <- data.frame(x = c(0.65, 0.25, 1.75, 0.75, 1.75), y = c(1.75, 1.4, 1.4, 1.05, 1.05))
Cuadros <- data.frame(x = c(1.35, 0.75, 1.25, 0.25, 1.25), y = c(1.75, 1.4, 1.4, 1.05, 1.05))

data.frame(x = c(3.1, 3.9, 3.5, 2.8, 4.2), y = c(1.0, 1.0, 0.2, 0.2, 0.2)) %>%
  
  ggplot(aes(x0 = x, y0 = y, r = 0.4)) +
  
  annotate(geom = 'rect', xmin = 0.4, xmax = 2.0, ymin = 1.9, ymax = 1.2, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  
  geom_point(aes(x = x, y = y), data = Puntos, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadros, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  #annotate(geom = 'rect', xmin = 3.05, xmax = 4.95, ymin = 0.05, ymax = 1.95, colour = 'black', fill = 'white', size = 0.2, linetype = 'solid') +
  
  #annotate(geom = 'segment', x = 1.95, xend = 3.05, y = 1.25, yend = 0.05, colour = 'black', size = 0.2, linetype = 'dashed') +
  #annotate(geom = 'segment', x = 0.55, xend = 3.35, y = 1.95, yend = 1.95, colour = 'black', size = 0.2, linetype = 'dashed') +
  
  geom_circle(aes(x0 = x, y0 = y, r = 0.25), colour = 'yellow', fill = 'yellow', alpha = 0.2, size = 0.4) +
  
  annotate(geom = 'segment', x = 3.03, xend = 3.17, y = 1.05, yend = 1.05, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 3.03, xend = 3.17, y = 0.95, yend = 0.95, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 3.1, y = 1.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 3.1, y = 0.9, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 3.83, xend = 3.97, y = 1.05, yend = 1.05, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 3.83, xend = 3.97, y = 0.95, yend = 0.95, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 3.9, y = 1.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 3.9, y = 0.9, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 3.43, xend = 3.57, y = 0.25, yend = 0.25, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 3.43, xend = 3.57, y = 0.15, yend = 0.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 3.5, y = 0.3, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 3.5, y = 0.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 2.73, xend = 2.87, y = 0.25, yend = 0.25, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 2.73, xend = 2.87, y = 0.15, yend = 0.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 2.8, y = 0.3, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 2.8, y = 0.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 4.13, xend = 4.27, y = 0.25, yend = 0.25, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 4.13, xend = 4.27, y = 0.15, yend = 0.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 4.2, y = 0.3, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.1, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 3.5, y = 1.0, label = 'X', size = 5.2, colour = 'black') +
  geom_curve(x = 3.5, xend = 3.5, y = 0.8, yend = 0.55, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  #annotate(geom = 'rect', xmin = 2.95, xmax = 3.45, ymin = -0.3, ymax = -0.1, colour = 'black', fill = 'white', size = 0.2, linetype = 'solid') +
  annotate(geom = 'text', x = 3.1, y = -0.2, label = TeX('$f_{4 5} = 1.0$'), size = 7.5) +
  
  #annotate(geom = 'rect', xmin = 3.55, xmax = 4.05, ymin = -0.3, ymax = -0.1, colour = 'black', fill = 'white', size = 0.2, linetype = 'solid') +
  annotate(geom = 'text', x = 3.9, y = -0.2, label = TeX('$f_{5 6} = 0.0$'), size = 7.5) +
  
  scale_x_continuous(limits = c(0.0, 4.45)) +
  scale_y_continuous(limits = c(-0.3, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:21")`.</FONT></p>

<P align="justify"><FONT SIZE=2>[2] `r Citet(mi_Bib, "cite:SJ")`.</FONT></p>

]

---
class: center

### Introducción

.panelset.sideways[

.panel[.panel-name[Mejor predictor lineal insesgado genómico] 

<P align="justify">La matriz **G** del GBLUP define la covarianza entre individuos en función de la similitud observada a nivel genómico, permitiendo con esto predicciones más precisas del valor genético<sup>1,2</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

Punto_1 <- data.frame(x = c(0.65, 1.75), y = c(1.75, 1.4))
Punto_2 <- data.frame(x = c(0.25, 0.75, 1.75), y = c(1.4, 1.05, 1.05))
Cuadro_1 <- data.frame(x = c(0.75, 1.25), y = c(1.4, 1.4))
Cuadro_2 <- data.frame(x = c(1.35, 0.25, 1.25), y = c(1.75, 1.05, 1.05))
Punto <- data.frame(x = 2.4, y = 1.6)
Cuadro <- data.frame(x = 2.4, y = 1.4)

ggplot() +
  
  geom_point(aes(x = x, y = y), data = Punto, size = 10.6, shape = 21, color = 'black', fill = 'black', alpha = 0.2) +
  geom_point(aes(x = x, y = y), data = Cuadro, size = 10.6, shape = 22, color = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 3.0, y = 1.5, label = 'Genotipados', size = 6.4, colour = 'black') +
  
  geom_point(aes(x = x, y = y), data = Punto_1, size = 17.6, shape = 21, color = 'black', fill = 'black', alpha = 0.2) +
  geom_point(aes(x = x, y = y), data = Punto_2, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadro_1, size = 17.6, shape = 22, color = 'black', fill = 'black', alpha = 0.2) +
  geom_point(aes(x = x, y = y), data = Cuadro_2, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.8, label = '0.13', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.8, label = '-0.40', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.8, label = '-0.40', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.8, label = '-0.67', size = 7.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.6, label = '-0.40', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.6, label = '1.20', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.6, label = '1.20', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.6, label = '2.00', size = 7.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.4, label = '-0.40', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.4, label = '1.20', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.4, label = '1.20', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.4, label = '2.00', size = 7.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.2, label = '-0.67', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.2, label = '2.00', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.2, label = '2.00', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.2, label = '3.33', size = 7.8, colour = 'black') +
  
  scale_x_continuous(limits = c(0.0, 3.55)) +
  scale_y_continuous(limits = c(0.1, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

<!--NOTA: a continuación se muestra parte de como se hizo el calculo de la matriz G, con base en el libro del profesor Agustin, considerando los individuos 1, 4, 5 y 6 como individuos genotipados, considerando que cada uno de ellos portan los siguientes alelos: ind_1 = Aa, ind_4 = AA, ind_5 = AA y ind_6 = aa. En este video (https://plant-breeding-genomics.extension.org/genomic-relationships-and-gblup/) también explican como hacer los calculos.-->

```{r, echo = FALSE, eval = FALSE}

W <- tribble(
  ~'',
  -0.25,
  0.75,
  0.75,
  1.25
  ) %>%
  as.matrix()

Wt <- t(W)

W_Wt <- W %*% Wt

G <- W_Wt * 2.13
```

<!--NOTA: otra forma de calcular la matriz G es por medio de una función presentada en el articulo "Genomic Best Linear Unbiased Prediction (gBLUP) for the Estimation of Genomic Breeding Values" de Samuel A. Clark and Julius van der Werf, en la página 328. Este articulo lo tengo en la carpeta "Articulos".-->

]

.panel[.panel-name[Selección genómica de múltiples pasos]

<P align="justify">El GBLUP es un método de varios pasos, con una evaluación genética con BLUP, seguidos de la extracción de pseudofenotipos, un análisis genómico y la creación de un índice que combina esta información<sup>3</sup>.</p>

```{r, echo = FALSE, out.width = "68%", fig.align = 'center', fig.showtext = TRUE}

data.frame(
  x = c(6.225, 6.225), y = c(6.7, 0.5)
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 0.5, xmax = 4.25, ymin = 6.0, ymax = 9.5, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 2.375, y = 7.75, label = 'A', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 0.0, y = 7.75, label = 'Matriz A', size = 7.6, angle = 90, colour = 'black') +
  
  geom_curve(x = 4.55, xend = 5.85, y = 7.75, yend = 7.75, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  annotate(geom = 'rect', xmin = 6.05, xmax = 8.65, ymin = 7.25, ymax = 8.25, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 7.35, y = 7.75, label = 'BLUP', size = 7.8, colour = 'black') +
  
  geom_curve(x = 8.85, xend = 10.325, y = 7.75, yend = 7.75, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  annotate(geom = 'text', x = 11.35, y = 7.75, label = 'EBV', size = 7.8, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 0.75, xmax = 3.25, ymin = 0.5, ymax = 2.5, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 2.0, y = 1.5, label = 'G', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.5, label = 'Matriz G', size = 7.6, angle = 90, colour = 'black') +
  
  geom_curve(x = 3.45, xend = 5.35, y = 1.5, yend = 1.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  annotate(geom = 'rect', xmin = 5.55, xmax = 8.15, ymin = 1.0, ymax = 2.0, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 6.85, y = 1.5, label = 'GBLUP', size = 7.8, colour = 'black') +
  
  geom_curve(x = 8.35, xend = 10.125, y = 1.5, yend = 1.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  annotate(geom = 'text', x = 11.45, y = 1.5, label = 'GEBV', size = 7.8, colour = 'black') +
  
  geom_curve(x = 11.15, xend = 6.85, y = 7.25, yend = 2.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  annotate(geom = 'text', x = 11.45, y = 5.75, label = 'Pseudo-', size = 7.8, colour = 'black') +
  annotate(geom = 'text', x = 11.45, y = 4.95, label = 'fenotipos', size = 7.8, colour = 'black') +
  
  scale_x_continuous(limits = c(-0.5, 14.0)) +
  scale_y_continuous(limits = c(-0.5, 10.5)) +
  
  theme_bw() +
  #labs(title = 'GBLUP') +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 11, face = 'bold'))
```

]

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:SJ")`.</FONT></p>

<P align="justify"><FONT SIZE=2>[2] `r Citet(mi_Bib, "cite:76")`.</FONT></p>

<P align="justify"><FONT SIZE=2>[3] `r Citet(mi_Bib, "cite:17")`.</FONT></p>

]

???

### Proceso de selección genómica

La **selección genómica** son una serie de métodos que usan numerosos marcadores de ADN, para predecir los valores genéticos:

```{r, echo = FALSE, out.width = "52%", fig.align = 'center', fig.showtext = TRUE}

data.frame(
  x = c(0.0:10.0), y = c(0.0:10.0)
  ) %>%
  ggplot(aes(x, y)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 3.0, xmax = 7.0, ymin = 7.0, ymax = 9.0, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 5.0, y = 8.355, label = 'Modelo de sele-', size = 7.0, colour = 'black') +
  annotate(geom = 'text', x = 5.0, y = 7.625, label = 'cción genómica', size = 7.0, colour = 'black') +
  geom_curve(x = 5.0, xend = 5.0, y = 6.8, yend = 6.0, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'twodash') +
  annotate(geom = 'text', x = 2.1, y = 8.5, label = 'Fenotipo', size = 6.4, colour = 'black') +
  geom_curve(x = 1.2, xend = 2.8, y = 8.2, yend = 8.2, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'solid') +
  annotate(geom = 'text', x = 2.1, y = 7.77, label = 'Genotipo', size = 6.4, colour = 'black') +
  geom_curve(x = 1.2, xend = 2.8, y = 7.47, yend = 7.47, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  annotate(geom = 'rect', xmin = 3.0, xmax = 7.0, ymin = 4.8, ymax = 5.8, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 5.0, y = 5.3, label = 'Predicción', size = 7.0, colour = 'black') +
  geom_curve(x = 5.0, xend = 5.0, y = 4.6, yend = 3.8, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'twodash') +
  annotate(geom = 'text', x = 2.1, y = 5.445, label = 'Genotipo', size = 6.4, colour = 'black') +
  geom_curve(x = 1.2, xend = 2.8, y = 5.145, yend = 5.145, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  annotate(geom = 'rect', xmin = 5.5, xmax = 8.5, ymin = 2.6, ymax = 3.6, colour = 'green', fill = 'green', alpha = 0.2) +
  annotate(geom = 'text', x = 7.0, y = 3.1, label = 'Selección', size = 7.0, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 4.0, xmax = 5.0, ymin = 2.0, ymax = 2.75, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 4.5, y = 2.25, label = 'GEV', family = 'gochi', size = 4.8, colour = 'black') +
  annotate(geom = 'text', x = 4.5, y = 3.35, label = 'Planta /', size = 5.4, colour = 'black') +
  annotate(geom = 'text', x = 4.5, y = 3.05, label = 'animal C', size = 5.4, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 2.75, xmax = 3.75, ymin = 2.0, ymax = 3.25, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 3.25, y = 2.25, label = 'GEV', family = 'gochi', size = 4.8, colour = 'black') +
  annotate(geom = 'text', x = 3.25, y = 3.85, label = 'Planta /', size = 5.4, colour = 'black') +
  annotate(geom = 'text', x = 3.25, y = 3.55, label = 'animal B', size = 5.4, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 2.5, ymin = 2.0, ymax = 2.5, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 2.0, y = 2.25, label = 'GEV', family = 'gochi', size = 4.8, colour = 'black') +
  annotate(geom = 'text', x = 2.0, y = 3.1, label = 'Planta /', size = 5.4, colour = 'black') +
  annotate(geom = 'text', x = 2.0, y = 2.8, label = 'animal A', size = 5.4, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 7.9, xmax = 8.5, ymin = 6.45, ymax = 9.55, colour = 'black', fill = 'white', linetype = 'twodash') +
  annotate(geom = 'text', x = 8.2, y = 8.0, label = 'Entrenamiento', angle = 90, size = 6.4, colour = 'black') +
  geom_curve(x = 7.7, xend = 7.3, y = 8.0, yend = 8.0, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  annotate(geom = 'rect', xmin = 7.9, xmax = 8.5, ymin = 4.5, ymax = 6.1, colour = 'black', fill = 'white', linetype = 'twodash') +
  annotate(geom = 'text', x = 8.2, y = 5.3, label = 'Prueba', angle = 90, size = 6.4, colour = 'black') +
  geom_curve(x = 7.7, xend = 7.3, y = 5.3, yend = 5.3, arrow = arrow(length = unit(0.1, 'inch')), size = 0.8, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  scale_x_continuous(limits = c(0.0, 10.0)) +
  scale_y_continuous(limits = c(2.0, 9.55)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 14, face = 'bold'))
```

### El problema con los pseudofenotipos

<P align="justify">Se considera que este proceso (de múltiples pasos) es poco eficiente, relativamente complicado y con algunos inconvenientes:</p>

.pull-left[

.content-box-red[

<P align="justify">Pueden existir individuos con fenotipos y pseudofenotipos.</p>

]

<br>

.content-box-yellow[

<P align="justify">Las precisiones de los pseudofenotipos son difíciles de obtener.</p>

]

]

.pull-right[

.content-box-yellow[

<P align="justify">La creación de pseudofenotipos puede ser ineficaz si los individuos tienen poca descendencia.</p>

]

]

---
class: center

### Introducción

.panelset.sideways[

.panel[.panel-name[GBLUP de un solo paso] 

<P align="justify">En el ssGBLUP, los fenotipos de los individuos genotipados y no genotipados se analizan conjuntamente para predecir los valores genéticos, a partir de la matriz **H**<sup>1</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE}

data.frame(
  x = 4.5, y = 0.5
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 4.25, ymin = 7.5, ymax = 9.5, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 2.875, y = 8.5, label = 'A', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 1.1, y = 8.5, label = 'Matriz A', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 7.0, xmax = 8.75, ymin = 7.75, ymax = 9.0, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 7.875, y = 8.375, label = 'G', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 6.6, y = 8.375, label = 'Matriz G', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 4.25, ymin = 3.75, ymax = 5.75, colour = 'black', fill = 'white', alpha = 0.2) +
  annotate(geom = 'rect', xmin = 2.3, xmax = 4.05, ymin = 3.95, ymax = 5.05, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 1.1, y = 4.75, label = 'Matriz H', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 4.1, xmax = 6.3, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 5.2, y = 1.8, label = 'ssGBLUP', size = 7.6, colour = 'black') +
  
  geom_curve(x = 6.4, xend = 7.5, y = 1.8, yend = 1.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  annotate(geom = 'text', x = 8.4, y = 1.8, label = 'GEBV', size = 7.6, colour = 'black') +
  
  geom_curve(x = 2.875, xend = 2.875, y = 7.3, yend = 6.05, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  geom_curve(x = 7.875, xend = 4.55, y = 7.45, yend = 4.75, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  geom_curve(x = 2.875, xend = 3.95, y = 3.45, yend = 2.1, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  scale_x_continuous(limits = c(0.45, 9.4)) +
  scale_y_continuous(limits = c(1.0, 9.9)) +
  
  theme_bw() +
  #labs(title = 'ssGBLUP') +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 11, face = 'bold'))
```

]
<!--
.panel[.panel-name[Matriz H e inversa de H]

<P align="justify">La estructura de la matriz **H** es:</p>

```{r, eval = FALSE, echo = FALSE, out.width = "90%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'Matriz_H.png'))
```

<P align="justify">Y las ecuaciones del modelo mixto para una evaluación genética de un solo paso son:</p>

```{r, eval = FALSE, echo = FALSE, out.width = "70%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'EMM_h.png'))
```

<P align="justify">La inversa de **H** es más simple que la estructura de **H**:</p>

```{r, eval = FALSE, echo = FALSE, out.width = "56%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'Matriz_Hinv.png'))
```

]
-->
.panel[.panel-name[BLUP vs ssGBLUP]

.pull-left[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE, message = FALSE}

Puntos <- data.frame(x = c(0.65, 0.25, 1.75, 0.75, 1.75), y = c(1.75, 1.4, 1.4, 1.05, 1.05))
Cuadros <- data.frame(x = c(1.35, 0.75, 1.25, 0.25, 1.25), y = c(1.75, 1.4, 1.4, 1.05, 1.05))

ggplot() +
  
  annotate(geom = 'rect', xmin = 0.0, xmax = 2.0, ymin = 1.9, ymax = 1.2, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 1.54, xmax = 2.06, ymin = 0.7, ymax = 0.9, colour = 'yellow', fill = 'yellow', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 2.74, xmax = 3.26, ymin = 0.3, ymax = 0.5, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  annotate(geom = 'rect', xmin = 3.34, xmax = 3.86, ymin = 0.3, ymax = 0.5, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  annotate(geom = 'rect', xmin = 3.94, xmax = 4.46, ymin = 0.3, ymax = 0.5, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 3.34, xmax = 3.86, ymin = 0.1, ymax = 0.3, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  annotate(geom = 'rect', xmin = 3.94, xmax = 4.46, ymin = 0.1, ymax = 0.3, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 3.94, xmax = 4.46, ymin = -0.1, ymax = 0.1, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  geom_point(aes(x = x, y = y), data = Puntos, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadros, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.8, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.8, label = '0.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.6, label = '0.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.6, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.4, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.2, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.0, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = -0.2, label = '1.00', size = 6.8, colour = 'black') +
  
  scale_x_continuous(limits = c(0.0, 4.46)) +
  scale_y_continuous(limits = c(-0.3, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())

mA <- tribble(
  ~'1', ~'2', ~'3', ~'4', ~'5', ~'6',
  NA, 0.0, 0.5, 0.5, 0.5, 0.5,
  0.0, NA, 0.5, 0.5, 0.5, 0.5,
  0.5, 0.5, NA, 0.5, 0.5, 0.5,
  0.5, 0.5, 0.5, NA, 0.5, 0.5,
  0.5, 0.5, 0.5, 0.5, NA, 0.5,
  0.5, 0.5, 0.5, 0.5, 0.5, NA
  ) %>%
  as.matrix()

mA <- melt(mA)

ggplot(data = mA, aes(x = value, after_stat(density))) +
  geom_histogram(colour = 'cyan', fill = 'cyan', alpha = 0.2, bins = 8) +
  labs(x = 'relación', y = ' ') +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 22, face = 'bold'),
    axis.title.x = element_text(size = 22, face = 'bold'),
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    )
```

]

.pull-right[

```{r Calculó de la matriz H, echo = FALSE, eval = FALSE}

# 1. Se crea el pedigrí ----

id <- seq(from = 1, to = 6, by = 1)
sire <- c(0, 0, 1, 1, 1, 1)
dam <- c(0, 0, 2, 2, 2, 2)

Pedigri <- tibble(id, sire, dam) %>%
  na_if(0) %>%
  mutate(
    Genotiped = case_when(
      Animal %in% c(1, 4:6) ~ TRUE,
      TRUE ~ FALSE
    )
  )

# 2. Se construye la matriz de relaciones genómicas (G) ----

Matriz_G <- Matrix(matrix(c(0.13, -0.40, -0.40, -0.67,
                            -0.40, 1.20, 1.20, 2.0,
                            -0.40, 1.20, 1.20, 2.0,
                            -0.67, 2.0, 2.0, 3.33),
                          nrow = 4))

# 3. Se coloca la función que calcula la matriz H ----

fn.mH <- function(ped, mG) { # Esta función recibe como argumentos los datos con estructura (id | sire | dam | Gen (TRUE/FALSE)) y la matriz de relaciones genómicas.
  
  # 1. Se calcula la matriz de relaciones aditivas con base en el pedigrí (A)
  
  ped_edit <- pedigreemm::editPed( # Esta función ordena el pedigrí. Luego la salida se usa posteriormente para crear un objeto "ped".
    sire = ped$sire,
    dam = ped$dam,
    label = ped$id
    )
  pedi <- pedigreemm::pedigree(
    sire = ped_edit$sire,
    dam = ped_edit$dam,
    label = ped_edit$label
  )
  Matrix_A <- pedigreemm::getA(ped = pedi) # El argumento debe ser un objeto de clase pedigree (el anteriormente creado). Esto dara la matriz de relaciones aditivas A.
 
  # 2. De lo anterior (Matriz_A) se extraen las partes correspondientes a individuos no genotipados (1) y genotipados (2)
  
  # Individuos no genotipados:
  A_11 <- Matrix_A[ped$Genotiped != 1, ped$Genotiped != 1] # Recordar que el R identifica a TRUE como 1 (de valor logico a número) y a FALSE como 0 (en este caso distinto de 1).
  # Individuos genotipados:
  A_22 <- Matrix_A[ped$Genotiped == 1, ped$Genotiped == 1]
  # Individuos no genotipados (en filas) y genotipados (en columnas):
  A_12 <- Matrix_A[ped$Genotiped != 1, ped$Genotiped == 1]
  # Transpuesta de la anterior (individuos no genotipados en columnas y genotipados en filas):
  A_21 <- t(A_12)
  
  # 3. Se coloca el nombre de las filas y y de las columnas de la matriz G según los individuos genotipados
  
  rownames(mG) <- ped$id[ped$Genotiped == 1]
  colnames(mG) <- ped$id[ped$Genotiped == 1]
  
  # 4. Teniendo todos los componentes de la matriz H, se procede a su construcción
  
  H_11 <- A_11 - (A_12 %*% solve(A_22) %*% A_21) + (A_12 %*% solve(A_22) %*% mG %*% solve(A_22) %*% A_21)
  H_12 <- A_12 %*% solve(A_22) %*% mG
  H_21 <- t(H_12)
  H_22 <- mG
  
  H_11_H_12 <- cbind(H_11, H_12)
  H_21_H_22 <- cbind(H_21, H_22)
  mH <- rbind(H_11_H_12, H_21_H_22)
  
  mH <- mH[order(as.numeric(rownames(mH))), order(as.numeric(colnames(mH)))]
  
  # 5. Finalmente se indica retornar la matriz H
  
  return(mH)
  }

# 4. Se usa la función para calcular la matriz H ----

fn.mH(ped = Pedigri, mG = Matriz_G)
```

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE, message = FALSE}

Punto_1 <- data.frame(x = c(0.65, 1.75), y = c(1.75, 1.4))
Punto_2 <- data.frame(x = c(0.25, 0.75, 1.75), y = c(1.4, 1.05, 1.05))
Cuadro_1 <- data.frame(x = c(0.75, 1.25), y = c(1.4, 1.4))
Cuadro_2 <- data.frame(x = c(1.35, 0.25, 1.25), y = c(1.75, 1.05, 1.05))
Punto <- data.frame(x = 2.4, y = 1.6)
Cuadro <- data.frame(x = 2.4, y = 1.4)

ggplot() +
  
  annotate(geom = 'rect', xmin = 0.0, xmax = 2.0, ymin = 1.9, ymax = 1.2, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 1.54, xmax = 2.06, ymin = 0.7, ymax = 0.9, colour = 'yellow', fill = 'yellow', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 2.74, xmax = 3.26, ymin = 0.3, ymax = 0.5, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  annotate(geom = 'rect', xmin = 3.34, xmax = 3.86, ymin = 0.3, ymax = 0.5, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  annotate(geom = 'rect', xmin = 3.94, xmax = 4.46, ymin = 0.3, ymax = 0.5, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 3.34, xmax = 3.86, ymin = 0.1, ymax = 0.3, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  annotate(geom = 'rect', xmin = 3.94, xmax = 4.46, ymin = 0.1, ymax = 0.3, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 3.94, xmax = 4.46, ymin = -0.1, ymax = 0.1, colour = 'red', fill = 'red', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  geom_point(aes(x = x, y = y), data = Punto, size = 10.6, shape = 21, color = 'black', fill = 'black', alpha = 0.2) +
  geom_point(aes(x = x, y = y), data = Cuadro, size = 10.6, shape = 22, color = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 3.0, y = 1.5, label = 'Genotipados', size = 6.4, colour = 'black') +
  
  geom_point(aes(x = x, y = y), data = Punto_1, size = 17.6, shape = 21, color = 'black', fill = 'black', alpha = 0.2) +
  geom_point(aes(x = x, y = y), data = Punto_2, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadro_1, size = 17.6, shape = 22, color = 'black', fill = 'black', alpha = 0.2) +
  geom_point(aes(x = x, y = y), data = Cuadro_2, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.8, label = '0.13', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.8, label = '-0.67', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.8, label = '-0.27', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.8, label = '-0.40', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.8, label = '-0.40', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.8, label = '-0.67', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.6, label = '-0.67', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.6, label = '3.73', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.6, label = '1.53', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.6, label = '2.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.6, label = '2.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.6, label = '3.33', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.4, label = '-0.27', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.4, label = '1.53', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.4, label = '1.13', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.4, label = '0.80', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.4, label = '0.80', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.4, label = '1.33', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.2, label = '-0.40', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.2, label = '2.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.2, label = '0.80', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.2, label = '1.20', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.2, label = '1.20', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.2, label = '2.00', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.0, label = '-0.40', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.0, label = '2.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.0, label = '0.80', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.0, label = '1.20', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.0, label = '1.20', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.0, label = '2.00', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = -0.2, label = '-0.67', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = -0.2, label = '3.33', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = -0.2, label = '1.33', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = -0.2, label = '2.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = -0.2, label = '2.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = -0.2, label = '3.33', size = 6.8, colour = 'black') +
  
  scale_x_continuous(limits = c(0.0, 4.46)) +
  scale_y_continuous(limits = c(-0.3, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())

mH <- tribble(
  ~'1', ~'2', ~'3', ~'4', ~'5', ~'6',
  NA, -0.67, -0.27, -0.40, -0.40, -0.67,
  -0.67, NA, 1.53, 2.00, 2.00, 3.33,
  -0.27, 1.53, NA, 0.80, 0.80, 1.33,
  -0.40, 2.00, 0.80, NA, 1.20, 2.00,
  -0.40, 2.00, 0.80, 1.20, NA, 2.00,
  -0.67, 3.33, 1.33, 2.00, 2.00, NA
  ) %>%
  as.matrix()

mH <- melt(mH)

ggplot(data = mH, aes(x = value, after_stat(density))) +
  geom_histogram(colour = 'cyan', fill = 'cyan', alpha = 0.2, bins = 8) +
  labs(x = 'relación', y = ' ') +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 22, face = 'bold'),
    axis.title.x = element_text(size = 22, face = 'bold'),
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    )
```

]

]

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:14")`.</FONT></p>

]

???

### Derivación del ssGBLUP

Conociendo que la matriz $H$ equivale a:

```{r, echo = FALSE, out.width = "32%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'ssGBLUP_1.png'))
```

El desarrollo de las ecuaciones que conducen a la matriz $H$ y su posterior uso dentro de las ecuaciones del modelo mixto se explican a continuación.

Partiendo de un modelo GBLUP en el que no se incluyen efectos fijos, $y = Zg + e$, un modelo en el cual se incluyen tanto individuos genotipados como no genotipados puede ser de la forma:

\begin{align*}
y = Z \begin{bmatrix} g_{1} \\ g_{2} \end{bmatrix} + e,
\end{align*}

donde $g_{1}$ corresponde a los individuos no genotipados, y $g_{2}$ a los individuos genotipados. Esto es, el vector efectos del marcador en el fenotipo o valores de cría ( $g$ ) es dividido en dos partes, una con los valores de cría de los individuos no genotipados ( $g_{1}$ ) y otra con los valores de cría de los individuos genotipados ( $g_{2}$ ).

Para estimar el parentesco genómico de los individuos no genotipados, sus valores de cría ( $g_{1}$ ) se predicen a partir de los valores de cría de los individuos que si lo están ( $g_{2}$ ), con base en la expresión:

\begin{align*}
g_{1} = cov(g_{1}, g_{2}) \times \left[ var(g_{2} \right]^{-1} \times g_{2} + e
\end{align*}

<!--
NOTA: para entender la expresión anterior se debe recordar la ecuación de regresión, la cual permite predecir una variable dependiente (y) en función de una variable independiente (x), o bien, predecir el valor aditivo de un toro en función del rendimiento lechero de sus hijas.

Para el caso del ssGBLUP, en el que se pretende predecir el valor aditivo de los individuos no genotipados (g1) en función de los individuos genotipados (g2), la ecuación de regresión, así como su expresión, serían de la siguiente forma:

\begin{align*}
b_{a_{1}, a_{2}} = \frac{\sigma_{a_{1}, a_{2}}}{\sigma_{a_{2}}^{2}} = \sigma_{a_{1}, a_{2}} \times \left[ \sigma_{a_{2}}^{2}^{-1} \right]
\end{align*}

Luego:

\begin{align*}
a_{1} = \sigma_{a_{1}, a_{2}} \times \left[ \sigma_{a_{2}}^{2} \right]^{-1} \times a_{2} + e
\end{align*}
-->

Sabiendo que la matriz de parentesco en base al pedigrí ( $A$ ) puede descomponerse como $\begin{bmatrix} A_{11} & A_{12} \\ A_{21} & A_{22} \end{bmatrix}$, donde $A_{11}$ y $A_{22}$ corresponden, respectivamente, a las matrices de parentesco en base al pedigrí de los individuos no genotipados y genotipados (o bien, la varianza de sus respectivos valores  de cría), y $A_{12}$ = $A_{21}$ corresponden a la matriz de parentesco en base al pedigrí entre individuos genotipados y no genotipados (o bien, la covarianza de sus valores de cría), la expresión anterior puede reescribirse de la siguiente forma:

\begin{align*}
g_{1} = A_{12} A_{22}^{-1} g_{2} + e
\end{align*}

Luego la expresión anterior en términos de varianza:

\begin{align*}
var(g_{1}) = var(A_{12} A_{22}^{-1} g_{2}) + var(e)
\end{align*}

Con un poco de algebra, finalmente la expresión correspondiente a $var(g_{1})$ es:

\begin{align*}
var(g_{1}) = A_{11} + A_{12} A_{22}^{-1} (G - A_{22}) A_{22}^{-1} A_{21}
\end{align*}

En relación a la expresión $var(g_{2})$, ésta sería igual a:

\begin{align*}
var(g_{2}) = G
\end{align*}

Por último, la expresión $cov(g_{1}, g_{2})$, equivaldría a:

\begin{align*}
cov(g_{1}, g_{2}) = cov(A_{12} A_{22}^{-1} g_{2} + e, g_{2})
\end{align*}

Qué con un poco de algebra, equilvaldría a:

\begin{align*}
cov(g_{1}, g_{2}) = A_{12} A_{22}^{-1} G
\end{align*}

Finalmente, la matriz que contiene las relaciones conjuntas de individuos genotipados y no genotipados sería:

```{r, echo = FALSE, out.width = "58%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'ssGBLUP_2.png'))
```

Una vez obtenida la matriz H, se obtiene su inversa ( $H^{-1}$ ), y el modelo de selección genómica se puede resolver mediante un modelo lineal mixto:

```{r, echo = FALSE, out.width = "40%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'EMM_h.png'))
```

---
class: center

### Objetivo

<P align="justify">Evaluar el interés de la selección genómica en un solo paso en plantas. Para ello, se evaluó el efecto de la cantidad de individuos genotipados y del número de marcadores en un diseño que combinó datos reales de arroz y simulaciones de dos tipos, hacia atrás (ancestros) y hacia delante (descendientes).</p>

---
class: center

### Materiales y métodos

<P align="justify"><b>Recurso vegetal y datos fenotípicos</b></p>

.pull-left[

<P align="justify">Se usaron los conjuntos de datos del [Rice SNP-Seek Database](http://iric.irri.org/), editados en un estudio previo<sup>1</sup>. Adicionalmente, se consideró un MAF inferior a 0.5.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'Arroz (Oryza sativa L.)',
  'Ploidía  →  2n = 24',
  'No. SNP  →  100.231'
  ) %>%
  gt()
```

<br>

<P align="justify">Se eligió la variedad Indica y el carácter tiempo de floración.</p>

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message=FALSE, warning=FALSE}

# 1. Se importan los archivos de datos.

m_dist <- read_table(here('Presentacion', 'Imagen', 'PCA_IND.mdist'), col_names = FALSE) %>%
  select(-X739)

ind <- tibble(id = read_table(here('Presentacion', 'Imagen', 'PCA_IND.mdist.id'), col_names = FALSE)[, 2]) # Se extrajo el nombre de los individuos.

esp <- read_delim(here('Presentacion', 'Imagen', 'all.phenotypes.csv'), delim = ',') %>%
  select(SNPsubsp) # Se extrajo el nombre de las especies (ADM, ARO, AUS, IND y JAP).

# 2. Se calculan los componentes principales en función del escalado multidimensional.

mds_poblacion <- cmdscale(d = m_dist, eig = TRUE, k = 5)

# 3. Se extraen los vectores propios.

vec_propios <- cbind(esp, ind, mds_poblacion$points)

# 4. Se calcula la proporción de la variación explicada por cada valor propio (por los dos primeros componentes principales).

porc_propio <- round(((mds_poblacion$eig) / sum(mds_poblacion$eig)) * 100, digits = 2)

porc_propio_2 <- porc_propio %>%
  as_tibble() %>%
  slice(1:12) %>%
  rowid_to_column(var = 'componente') %>%
  mutate(componente = as.factor(componente))

# 5. Se visualizan los resultados mediante componentes principales.

graf_bar <- ggplot(data = porc_propio_2, aes(x = componente, y = value)) +
  geom_bar(stat = 'identity', color = 'gray34', fill = 'gray34', alpha = 0.4) +
  labs(
   x = 'Componente principal',
   y = 'Variación acumulada (%)'
   ) +
  theme(
    axis.text.x = element_text(face = 'bold', size = 7),
    axis.text.y = element_text(face = 'bold', size = 7, angle = 90),
    axis.title = element_text(face = 'bold', size = 8)
    )

ggplot(data = vec_propios, mapping = aes(x = `1`, y = `2`, color = SNPsubsp, fill = SNPsubsp)) +
  geom_point(alpha = 0.2, size = 4.4) +
  stat_ellipse(aes(color = SNPsubsp), type = 't') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  scale_color_manual(values = c('yellow', 'black', 'red', 'cyan', 'green')) +
  scale_fill_manual(values = c('yellow', 'black', 'red', 'cyan', 'green')) +
  labs(
   x = paste0('Primer componente (',porc_propio[1],' %)'),
   y = paste0('Segundo componente (',porc_propio[2], '%)'),
   col = 'Variedades de arroz', fill = 'Variedades de arroz'
   ) +
  theme_bw() +
  theme(
    legend.background = element_rect(color = 'gray34', fill = 'white', size = 0.4, linetype = 'dashed'),
    legend.position = c(0.18, 0.24),
    legend.text = element_text(size = 11),
    legend.title = element_text(face = 'bold', size = 11),
    axis.text = element_text(face = 'bold', size = 11),
    axis.title = element_text(face = 'bold', size = 11)
    )
```

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:26")`.</FONT></p>

]

---
class: center

### Materiales y métodos

<P align="justify"><b>Predicción basada en información de pedigrí e información genómica</b></p>

.pull-left[

<P align="justify">El modelo general fue:</p>

\begin{equation}
y = \mu + Zg + e
\end{equation}

<br>

<P align="justify">Los supuestos del modelo son:</p>

+ <P align="justify">BLUP $\rightarrow$ $g$ ~ $N(0, A \sigma^{2}_{g})$</p>

+ <P align="justify">ssGBLUP $\rightarrow$ $g$ ~ $N(0, H \sigma^{2}_{g})$</p>

]

.pull-right[

<P align="justify">La matriz **H** fue construida como:</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'Matriz_H.png'))
```

<P align="justify">La matriz **G** se obtuvo a partir del método de `r Citet(mi_Bib, "cite:39")`:</p>

\begin{equation}
G =\frac{XX'}{2 \sum_{j=1}^{nSNP} p_{j} (1 - p_{j})}
\end{equation}

]

--

<br>

<P align="justify">La heredabilidad en sentido estricto se estimó como:</p>

\begin{equation}
h^{2} = \frac{\sigma^{2}_{g}}{\sigma^{2}_{g} + \sigma^{2}_{e}}
\end{equation}

---
class: center

### Materiales y métodos

<P align="justify"><b>Precisión de la predición mediante simulación ancestral</b></p>

.pull-left[

<P align="justify">Se utilizó la metodología del software Molcoanc<sup>1</sup> con el fin de construir tres pedigríes.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'Pedigrí', ~'Núm. de individuos',
  'Pedigrí 1', '300 + 451 = 751',
  'Pedigrí 2', '1210 + 451 = 1661',
  'Pedigrí 3', '2000 + 451 = 2451'
  ) %>%
  gt()
```

<P align="justify">Este proceso se replicó diez veces para cada pedigrí.</p>

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE, message = FALSE}

Puntos <- data.frame(x = c(1.0, 1.5, 0.75, 1.25, 1.75, 0.5, 1.0, 1.5, 2.0, 0.25, 0.75, 1.25, 1.75, 2.25), y = c(c(2.8, 2.8, 1.5, 1.5, 1.5, 0.2, 0.2, 0.2, 0.2, -1.1, -1.1, -1.1, -1.1, -1.1)))

data.frame(
  x = 1.0, y = 1.0
  ) %>%
  ggplot() +
  
  annotate(geom = 'rect', xmin = -0.4, xmax = 2.5, ymin = -1.7, ymax = 3.4, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  geom_point(aes(x = x, y = y), data = Puntos, size = 5.6, shape = 21, color = 'black', fill = 'black', alpha = 0.8) +
  geom_curve(x = 0.05, xend = 0.05, y = -1.1, yend = 2.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  annotate(geom = 'text', x = -0.3, y = 0.85, label = 'Creación de antepasados de indivi-', size = 5.2, colour = 'black', angle = 90) +
  annotate(geom = 'text', x = -0.1, y = 0.85, label = 'duos genotipados mediante simulación', size = 5.2, colour = 'black', angle = 90) +
  
  annotate(geom = 'segment', x = 1.0, xend = 0.75, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.5, xend = 0.75, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.0, xend = 1.25, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.5, xend = 1.25, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.5, xend = 1.75, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 0.75, xend = 0.5, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.25, xend = 1.0, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.25, xend = 2.0, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.75, xend = 2.0, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.75, xend = 1.5, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.5, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 0.75, xend = 1.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.25, xend = 1.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.75, xend = 2.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 2.25, xend = 2.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  
  geom_curve(x = 2.6, xend = 3.0, y = 2.8, yend = 2.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 3.1, xmax = 4.1, ymin = 3.0, ymax = 1.8, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 3.6, y = 2.4, label = 'A', size = 10.4, colour = 'black') +
  
  scale_x_continuous(limits = c(-0.4, 4.2)) +
  scale_y_continuous(limits = c(-1.8, 3.4)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:24")`.</FONT></p>

]

---
class: center

### Materiales y métodos

<P align="justify"><b>Precisión de la predición mediante simulación ancestral</b></p>

.pull-left[

<P align="justify">Se uso la $cor(y, \hat{y})$ como medida de predictibilidad, en 48 individuos.</p>

<br>

<P align="justify">Se usaron diferentes subconjuntos de datos con la siguientes características:</p>

<P align="justify">a) Distinta cantidad de individuos genotipados.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'100 + 48 = 148', ~'250 + 48 = 298 ', ~'451'
  ) %>%
  gt()
```

<P align="justify">b) Diferentes densidades de marcadores.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'1.000', ~'10.000 ', ~'100.000'
  ) %>%
  gt()
```

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE, message = FALSE}

data.frame(
  x = 1.0, y = 1.0
  ) %>%
  ggplot() +
  
  annotate(geom = 'text', x = 0.5, y = 3.5, label = 'Ind. genotipados', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 1.5, y = 3.5, label = '148', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 2.5, y = 3.5, label = '298', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 3.5, y = 3.5, label = '298', size = 6.2, colour = 'black') +
  
  annotate(geom = 'text', x = 0.5, y = 2.0, label = 'Densidad de SNP', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 1.5, y = 2.0, label = '1.000', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 2.5, y = 2.0, label = '10.000', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 3.5, y = 2.0, label = '100.000', size = 6.2, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 1.1, xmax = 1.9, ymin = 1.0, ymax = 0.4, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 1.5, y = 0.7, label = 'G', size = 9.4, colour = 'black') +
  annotate(geom = 'rect', xmin = 2.1, xmax = 2.9, ymin = 1.0, ymax = 0.4, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 2.5, y = 0.7, label = 'G', size = 9.4, colour = 'black') +
  annotate(geom = 'rect', xmin = 3.1, xmax = 3.9, ymin = 1.0, ymax = 0.4, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 3.5, y = 0.7, label = 'G', size = 9.4, colour = 'black') +
  
  geom_curve(x = 1.5, xend = 1.5, y = 3.3, yend = 2.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 1.5, xend = 2.5, y = 3.3, yend = 2.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 1.5, xend = 3.5, y = 3.3, yend = 2.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  
  geom_curve(x = 1.5, xend = 1.5, y = 1.8, yend = 1.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 2.5, xend = 2.5, y = 1.8, yend = 1.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 3.5, xend = 3.5, y = 1.8, yend = 1.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  
  scale_x_continuous(limits = c(0.0, 3.9)) +
  scale_y_continuous(limits = c(0.4, 3.6)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

---
class: center

### Materiales y métodos

<P align="justify"><b>Precisión de la predición mediante simulación ancestral</b></p>

.pull-left[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE}

data.frame(
  x = 4.5, y = 0.5
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 4.25, ymin = 7.5, ymax = 9.5, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 2.875, y = 8.5, label = 'A', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 1.1, y = 8.5, label = 'Matriz A', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 7.0, xmax = 8.75, ymin = 7.75, ymax = 9.0, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 7.875, y = 8.375, label = 'G', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 6.6, y = 8.375, label = 'Matriz G', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 6.0, xmax = 8.75, ymin = 3.75, ymax = 5.75, colour = 'black', fill = 'white', alpha = 0.2) +
  annotate(geom = 'rect', xmin = 6.8, xmax = 8.55, ymin = 3.95, ymax = 5.05, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 5.6, y = 4.75, label = 'Matriz H', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 6.275, xmax = 8.475, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 7.375, y = 1.8, label = 'ssGBLUP', size = 7.6, colour = 'black') +
  geom_curve(x = 7.375, xend = 7.375, y = 3.45, yend = 2.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 1.55, xmax = 3.75, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 2.65, y = 1.8, label = 'BLUP', size = 7.6, colour = 'black') +
  
  geom_curve(x = 3.0, xend = 5.2, y = 7.3, yend = 4.9, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  geom_curve(x = 7.875, xend = 7.875, y = 7.45, yend = 6.05, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  geom_curve(x = 2.65, xend = 2.65, y = 7.3, yend = 2.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  scale_x_continuous(limits = c(0.45, 9.4)) +
  scale_y_continuous(limits = c(1.0, 9.9)) +
  
  theme_bw() +
  #labs(title = 'ssGBLUP') +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 11, face = 'bold'))
```

]

--

.pull-right[

<P align="justify">Se usaron los paquetes `BGLR`<sup>1</sup> y `lme4GS`<sup>2</sup> del `R` para estimar los componentes de varianza y los valores fenotípicos predichos, permitiendo así la predicción mediante procedimientos Bayesianos y REML.</p>

<br>

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:50")`.</FONT></p>
<P align="justify"><FONT SIZE=2>[2] `r Citet(mi_Bib, "cite:51")`.</FONT></p>

]

]

---
class: center

### Materiales y métodos

<P align="justify"><b>Precisión de la predición mediante simulación de descendientes</b></p>

.pull-left[

<P align="justify">Se realizó un estudio de simulación usando el módulo de `Python` `SeqBreed`<sup>1</sup>.</p>

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE}

Puntos <- data.frame(x = c(9.4, 16.2, 13.6, 17.2, 12.0), y = c(8.2, 6.6, 3.35, -0.25, -6.6))

data.frame(
  x = 10.0, y = -7.8
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  #annotate(geom = 'rect', xmin = 7.8, xmax = 12.2, ymin = 4.3, ymax = 5.1, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 10.0, y = 4.7, label = 'Población fundadora', size = 7.6, colour = 'black') +
    
  #annotate(geom = 'rect', xmin = 3.6, xmax = 8.0, ymin = 7.2, ymax = 8.0, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 5.8, y = 7.6, label = 'Genotipos fundadores', size = 7.6, colour = 'black') +
  geom_curve(x = 5.8, xend = 7.6, y = 6.9, yend = 5.35, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  
  #annotate(geom = 'rect', xmin = 12.0, xmax = 16.4, ymin = 6.4, ymax = 8.0, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 14.2, y = 7.6, label = 'Localización de QTN', size = 7.6, colour = 'black') +
  annotate(geom = 'text', x = 14.2, y = 6.8, label = 'y efectos', size = 7.6, colour = 'black') +
  geom_curve(x = 14.2, xend = 12.4, y = 6.1, yend = 5.35, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  
  geom_curve(x = 10.0, xend = 10.0, y = 4.1, yend = 1.6, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  geom_curve(x = 10.8, xend = 10.2, y = 2.75, yend = 2.75, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  #annotate(geom = 'rect', xmin = 11.0, xmax = 13.2, ymin = 2.35, ymax = 3.15, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 12.1, y = 2.75, label = 'Pedigrí', size = 7.6, colour = 'black') +
  
  #annotate(geom = 'rect', xmin = 8.0, xmax = 12.0, ymin = 0.5, ymax = 1.3, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 10.0, y = 0.9, label = 'Población', size = 7.6, colour = 'black') +
  
  geom_curve(x = 10.0, xend = 10.0, y = 0.3, yend = -2.0, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  geom_curve(x = 10.8, xend = 10.2, y = -0.75, yend = -0.75, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  #annotate(geom = 'rect', xmin = 11.0, xmax = 14.6, ymin = -1.25, ymax = -0.35, colour = 'black', fill = 'black', alpha = 0.2) +
  annotate(geom = 'text', x = 13.8, y = -0.75, label = 'Densidad de SNP', size = 7.6, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 9.1, xmax = 10.8, ymin = -4.0, ymax = -2.2, colour = 'black', fill = 'white') +
  annotate(geom = 'rect', xmin = 9.7, xmax = 10.7, ymin = -3.75, ymax = -2.9, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 8.6, y = -3.1, label = 'Matriz H', size = 7.6, angle = 90, colour = 'black') +
  
  geom_curve(x = 10.0, xend = 10.0, y = -4.2, yend = -6.3, arrow = arrow(length = unit(0.1, 'inch')), size = 0.5, color = 'black', curvature = 0.0) +
  annotate(geom = 'text', x = 10.0, y = -7.0, label = 'ssGBLUP', size = 7.6, colour = 'black') +
  annotate(geom = 'text', x = 10.0, y = -7.8, label = 'BLUP', size = 7.6, colour = 'black') +
  
  geom_point(aes(x = x, y = y), data = Puntos, size = 7.4, shape = 16, color = 'gray84') +
  annotate(geom = 'text', x = 9.4, y = 8.2, label = '1', family = 'gochi', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 16.2, y = 6.6, label = '2', family = 'gochi', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 13.6, y = 3.35, label = '3', family = 'gochi', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 17.2, y = -0.25, label = '4', family = 'gochi', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 12.0, y = -6.6, label = '5', family = 'gochi', size = 6.8, colour = 'black') +
    
  scale_x_continuous(limits = c(2.5, 17.5)) +
  scale_y_continuous(limits = c(-8.3, 8.5)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:47")`.</FONT></p>

]

---
class: center

### Resultados



---

### Bibliografía

```{r refs 1, echo=FALSE, results="asis"}

PrintBibliography(bib = mi_Bib, start = 1, end = 6)
```

---

### Bibliografía

```{r refs 2, echo=FALSE, results="asis"}

PrintBibliography(bib = mi_Bib, start = 7, end = 11)
```

---

class: center, middle, inverse

<span class = "cc">
<!-- Key: a: SA, b: BY, c: CC Circle, d: ND, n: NC, m: Sampling, s: Share, r: Remix, C: CC Full Logo -->
C bn
</span>

[Licencia CC BY-NC](https://creativecommons.org/licenses/by-nc/4.0/)
