---
title: "Evaluación de la selección genómica de un solo paso en arroz"
output:
  xaringan::moon_reader:
    css: [default, css/cc-fonts.css, css/Cajitas.css]
    lib_dir: libs
    seal: false
    nature:
      #titleSlideClass: ["center", "top", "my-title"]
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      slideNumberFormat: "%current%<br>"
---
class: center

```{r Paquetes usados, general y bibliografia, load_refs, eval = TRUE, echo = FALSE, include = FALSE, cache = FALSE}

# Paquetes usados ----

pacman::p_load(metathis, xaringanExtra, fontawesome, RefManageR, here, tidyverse, latex2exp, showtext, ggforce, magick, pedigreemm, reshape2, gt) # Aquí (https://pkg.garrickadenbuie.com/xaringanExtra/#/) y aquí (https://github.com/gadenbuie/xaringanExtra) explican las muchas opciones de usar el paquete xaringanExtra.
font_add_google('Gochi Hand', 'gochi')
showtext_auto()

# Opciones generales de la presentación ----

workshop_day <- format(as.Date("2022-06-01"), format = "%B/%d/%Y")

# Habilita la capacidad de mostrar todas las diapositivas en una vista general de mosaico presionando "o"
xaringanExtra::use_tile_view()
xaringanExtra::use_panelset()
xaringanExtra::use_clipboard()
xaringanExtra::use_share_again()
xaringanExtra::style_share_again(share_buttons = "all")
xaringanExtra::use_extra_styles(
  hover_code_line = TRUE,
  mute_unhighlighted_code = FALSE
)

# Opciones de bibliografía ----

options(htmltools.dir.version = FALSE)

BibOptions(check.entries = FALSE, bib.style = "authoryear", style = "markdown",
           dashed = TRUE)

mi_Bib <- ReadBib(here('Presentacion', 'Bibliografia', 'mi_Bib.bib'), check = FALSE)
```

```{r metathis, eval = TRUE, echo = FALSE}

meta() %>%
  meta_general(
    description = "Presentación TFM",
    generator = "xaringan and remark.js"
  ) %>%
  meta_name('github-repo' = 'Leo4Luffy/Presentacion_TFM') %>% 
  meta_social(
    title = "Presentación del trabajo de fin de máster",
    url = "https://leo-tfm.netlify.app/",
    og_type = "website",
    og_author = "Jorge Leonardo López Martínez",
    twitter_card_type = "summary_large_image",
    twitter_creator = "@leo4luffy"
  )
```

```{r xaringanExtra-share-again, eval = FALSE, echo = FALSE}

xaringanExtra::use_share_again()
```

###.fancy[`r rmarkdown::metadata$title`] 

<!--###.fancy[`r rmarkdown::metadata$subtitle`]-->

<br>
<br>

<P align="justify"> **Estudiante:** Jorge Leonardo López Martínez <p>
<!--<P align="justify">[`r fa(name = "github", fill = "black")` Leo4Luffy](https://github.com/Leo4Luffy)<p>-->

<br>

<P align="justify">**Director:** Miguel Pérez-Enciso<p>
<!--<P align="justify">[`r fa(name = "github", fill = "black")` miguelperezenciso](https://github.com/miguelperezenciso)<p>-->


<br>
<br>

<P align="justify">`r workshop_day`<p>

---
class: center

### Introducción

.panelset.sideways[

.panel[.panel-name[Mejor predictor lineal insesgado]

<P align="justify">El BLUP es un método que se utiliza para predecir el valor genético de un individuo. Para ello, se usa una matriz de parentesco estimada a partir de la información del pedigrí<sup>1</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

data.frame(
  x = 4.5, y = 0.5
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 4.25, ymin = 3.75, ymax = 5.75, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 2.875, y = 4.75, label = 'A', size = 9.4, colour = 'black') +
  annotate(geom = 'text', x = 1.1, y = 4.75, label = 'Matriz A', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 4.1, xmax = 6.3, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 5.2, y = 1.8, label = 'BLUP', size = 7.6, colour = 'black') +
  
  geom_curve(x = 6.4, xend = 7.5, y = 1.8, yend = 1.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  annotate(geom = 'text', x = 8.4, y = 1.8, label = 'EBV', size = 7.6, colour = 'black') +
  
  geom_curve(x = 2.875, xend = 3.95, y = 3.45, yend = 2.1, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  
  scale_x_continuous(limits = c(0.45, 9.4)) +
  scale_y_continuous(limits = c(1.0, 6.3)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 11, face = 'bold'))
```

]

.panel[.panel-name[Matriz de parentesco- I]

<P align="justify">En la matriz **A**, los elementos fuera de la diagonal indican la relación entre individuos<sup>2</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

Puntos <- data.frame(x = c(0.65, 0.25, 1.75, 0.75, 1.75), y = c(1.75, 1.4, 1.4, 1.05, 1.05))
Cuadros <- data.frame(x = c(1.35, 0.75, 1.25, 0.25, 1.25), y = c(1.75, 1.4, 1.4, 1.05, 1.05))

ggplot() +
  
  annotate(geom = 'rect', xmin = 0.0, xmax = 2.0, ymin = 1.9, ymax = 1.2, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 3.35, xmax = 3.85, ymin = 0.1, ymax = 0.3, colour = 'yellow', fill = 'yellow', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  annotate(geom = 'rect', xmin = 3.95, xmax = 4.45, ymin = -0.1, ymax = 0.1, colour = 'yellow', fill = 'yellow', size = 0.4, linetype = 'dashed', alpha = 0.4) +
  
  geom_point(aes(x = x, y = y), data = Puntos, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadros, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.8, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.8, label = '0.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.8, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.6, label = '0.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.6, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.6, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.4, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.4, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.2, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.2, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = 0.0, label = '1.00', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.0, label = '0.50', size = 6.8, colour = 'black') +
  
  annotate(geom = 'text', x = 1.2, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 1.8, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 2.4, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.0, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 3.6, y = -0.2, label = '0.50', size = 6.8, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = -0.2, label = '1.00', size = 6.8, colour = 'black') +
  
  scale_x_continuous(limits = c(0.0, 4.45)) +
  scale_y_continuous(limits = c(-0.3, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

.panel[.panel-name[Matriz de parentesco- II]

<P align="justify">El parentesco que proviene al usar la matriz **A** es un parentesco esperado, lo cual no refleja el porcentaje real de genes idénticos entre individuos emparentados<sup>1</sup>.</p>

```{r, echo = FALSE, out.width = "70%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE}

Puntos <- data.frame(x = c(0.65, 0.25, 1.75, 0.75, 1.75), y = c(1.75, 1.4, 1.4, 1.05, 1.05))
Cuadros <- data.frame(x = c(1.35, 0.75, 1.25, 0.25, 1.25), y = c(1.75, 1.4, 1.4, 1.05, 1.05))

data.frame(x = c(3.1, 3.9, 3.5, 2.8, 4.2), y = c(1.0, 1.0, 0.2, 0.2, 0.2)) %>%
  
  ggplot(aes(x0 = x, y0 = y, r = 0.4)) +
  
  annotate(geom = 'rect', xmin = 0.4, xmax = 2.0, ymin = 1.9, ymax = 1.2, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  
  geom_point(aes(x = x, y = y), data = Puntos, size = 17.6, shape = 21, color = 'black', fill = 'white', alpha = 0.5) +
  geom_point(aes(x = x, y = y), data = Cuadros, size = 17.6, shape = 22, color = 'black', fill = 'white', alpha = 0.5) +
  
  annotate(geom = 'segment', x = 0.85, xend = 1.15, y = 1.75, yend = 1.75, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.0, xend = 1.0, y = 1.75, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 1.75, y = 1.6, yend = 1.6, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.6, yend = 1.5, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'segment', x = 1.75, xend = 1.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 1.25, xend = 1.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.75, xend = 0.75, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 0.25, xend = 0.25, y = 1.3, yend = 1.15, colour = 'black', size = 0.8, linetype = 'solid') +
  
  annotate(geom = 'text', x = 0.65, y = 1.75, label = '1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.35, y = 1.75, label = '2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.4, label = '3', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.4, label = '4', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.4, label = '5', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.4, label = '6', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.25, y = 1.05, label = '7', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 0.75, y = 1.05, label = '8', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.25, y = 1.05, label = '9', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 1.75, y = 1.05, label = '10', family = 'gochi', size = 7.2, colour = 'black') +
  
  #annotate(geom = 'rect', xmin = 3.05, xmax = 4.95, ymin = 0.05, ymax = 1.95, colour = 'black', fill = 'white', size = 0.2, linetype = 'solid') +
  
  #annotate(geom = 'segment', x = 1.95, xend = 3.05, y = 1.25, yend = 0.05, colour = 'black', size = 0.2, linetype = 'dashed') +
  #annotate(geom = 'segment', x = 0.55, xend = 3.35, y = 1.95, yend = 1.95, colour = 'black', size = 0.2, linetype = 'dashed') +
  
  geom_circle(aes(x0 = x, y0 = y, r = 0.25), colour = 'yellow', fill = 'yellow', alpha = 0.2, size = 0.4) +
  
  annotate(geom = 'segment', x = 3.03, xend = 3.17, y = 1.05, yend = 1.05, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 3.03, xend = 3.17, y = 0.95, yend = 0.95, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 3.1, y = 1.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 3.1, y = 0.9, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 3.83, xend = 3.97, y = 1.05, yend = 1.05, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 3.83, xend = 3.97, y = 0.95, yend = 0.95, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 3.9, y = 1.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 3.9, y = 0.9, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 3.43, xend = 3.57, y = 0.25, yend = 0.25, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 3.43, xend = 3.57, y = 0.15, yend = 0.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 3.5, y = 0.3, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 3.5, y = 0.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 2.73, xend = 2.87, y = 0.25, yend = 0.25, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 2.73, xend = 2.87, y = 0.15, yend = 0.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 2.8, y = 0.3, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 2.8, y = 0.1, label = 'A1', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'segment', x = 4.13, xend = 4.27, y = 0.25, yend = 0.25, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'segment', x = 4.13, xend = 4.27, y = 0.15, yend = 0.15, colour = 'black', size = 0.8, linetype = 'solid') +
  annotate(geom = 'text', x = 4.2, y = 0.3, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  annotate(geom = 'text', x = 4.2, y = 0.1, label = 'A2', family = 'gochi', size = 7.2, colour = 'black') +
  
  annotate(geom = 'text', x = 3.5, y = 1.0, label = 'X', size = 5.2, colour = 'black') +
  geom_curve(x = 3.5, xend = 3.5, y = 0.8, yend = 0.55, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  #annotate(geom = 'rect', xmin = 2.95, xmax = 3.45, ymin = -0.3, ymax = -0.1, colour = 'black', fill = 'white', size = 0.2, linetype = 'solid') +
  annotate(geom = 'text', x = 3.1, y = -0.2, label = TeX('$f_{4 5} = 1.0$'), size = 7.5) +
  
  #annotate(geom = 'rect', xmin = 3.55, xmax = 4.05, ymin = -0.3, ymax = -0.1, colour = 'black', fill = 'white', size = 0.2, linetype = 'solid') +
  annotate(geom = 'text', x = 3.9, y = -0.2, label = TeX('$f_{5 6} = 0.0$'), size = 7.5) +
  
  scale_x_continuous(limits = c(0.0, 4.45)) +
  scale_y_continuous(limits = c(-0.3, 1.9)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:21")`.</FONT></p>

<P align="justify"><FONT SIZE=2>[2] `r Citet(mi_Bib, "cite:SJ")`.</FONT></p>


]

---
class: center

### Introducción

.panelset.sideways[

.panel[.panel-name[Mejor predictor lineal insesgado genómico] 

]

]

---
class: center

### Materiales y métodos (parciales)

<P align="justify"><b>Recurso vegetal y datos fenotípicos</b></p>

.pull-left[

<P align="justify">Se usaron los conjuntos de datos del [Rice SNP-Seek Database](http://iric.irri.org/), editados en un estudio previo<sup>1</sup>. Adicionalmente, se consieró un MAF inferior a 0.5.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'Arroz (Oryza sativa L.)',
  'Ploidía  →  2n = 24',
  'No. SNP  →  100.231'
  ) %>%
  gt()
```

<br>

<P align="justify">Se eligió la variedad indica y el carácter tiempo de floración.</p>

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message=FALSE, warning=FALSE}

# 1. Se importan los archivos de datos.

m_dist <- read_table(here('Presentacion', 'Imagen', 'PCA_IND.mdist'), col_names = FALSE) %>%
  select(-X739)

ind <- tibble(id = read_table(here('Presentacion', 'Imagen', 'PCA_IND.mdist.id'), col_names = FALSE)[, 2]) # Se extrajo el nombre de los individuos.

esp <- read_delim(here('Presentacion', 'Imagen', 'all.phenotypes.csv'), delim = ',') %>%
  select(SNPsubsp) # Se extrajo el nombre de las especies (ADM, ARO, AUS, IND y JAP).

# 2. Se calculan los componentes principales en función del escalado multidimensional.

mds_poblacion <- cmdscale(d = m_dist, eig = TRUE, k = 5)

# 3. Se extraen los vectores propios.

vec_propios <- cbind(esp, ind, mds_poblacion$points)

# 4. Se calcula la proporción de la variación explicada por cada valor propio (por los dos primeros componentes principales).

porc_propio <- round(((mds_poblacion$eig) / sum(mds_poblacion$eig)) * 100, digits = 2)

porc_propio_2 <- porc_propio %>%
  as_tibble() %>%
  slice(1:12) %>%
  rowid_to_column(var = 'componente') %>%
  mutate(componente = as.factor(componente))

# 5. Se visualizan los resultados mediante componentes principales.

graf_bar <- ggplot(data = porc_propio_2, aes(x = componente, y = value)) +
  geom_bar(stat = 'identity', color = 'gray34', fill = 'gray34', alpha = 0.4) +
  labs(
   x = 'Componente principal',
   y = 'Variación acumulada (%)'
   ) +
  theme(
    axis.text.x = element_text(face = 'bold', size = 7),
    axis.text.y = element_text(face = 'bold', size = 7, angle = 90),
    axis.title = element_text(face = 'bold', size = 8)
    )

ggplot(data = vec_propios, mapping = aes(x = `1`, y = `2`, color = SNPsubsp, fill = SNPsubsp)) +
  geom_point(alpha = 0.2, size = 4.4) +
  stat_ellipse(aes(color = SNPsubsp), type = 't') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  scale_color_manual(values = c('yellow', 'black', 'red', 'cyan', 'green')) +
  scale_fill_manual(values = c('yellow', 'black', 'red', 'cyan', 'green')) +
  labs(
   x = paste0('Primer componente (',porc_propio[1],' %)'),
   y = paste0('Segundo componente (',porc_propio[2], '%)'),
   col = 'Variedades de arroz', fill = 'Variedades de arroz'
   ) +
  theme_bw() +
  theme(
    legend.background = element_rect(color = 'gray34', fill = 'white', size = 0.4, linetype = 'dashed'),
    legend.position = c(0.18, 0.24),
    legend.text = element_text(size = 11),
    legend.title = element_text(face = 'bold', size = 11),
    axis.text = element_text(face = 'bold', size = 11),
    axis.title = element_text(face = 'bold', size = 11)
    )
```

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:26")`.</FONT></p>

]

---
class: center

### Materiales y métodos (parciales)

<P align="justify"><b>Predicción basada en información de pedigrí e información genómica</b></p>

.pull-left[

<P align="justify">El modelo general fue:</p>

\begin{equation}
y = \mu + Zg + e
\end{equation}

<br>

<P align="justify">Los supuestos del modelo son:</p>

+ <P align="justify">BLUP $\rightarrow$ $g$ ~ $N(0, A \sigma^{2}_{g})$</p>

+ <P align="justify">ssGBLUP $\rightarrow$ $g$ ~ $N(0, H \sigma^{2}_{g})$</p>

]

.pull-right[

<P align="justify">La matriz **H** fue construida como:</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE, warning = FALSE}

image_read(here('Presentacion', 'Imagen', 'Matriz_H.png'))
```

<P align="justify">La matriz **G** se obtuvo a partir del método de `r Citet(mi_Bib, "cite:39")`:</p>

\begin{equation}
G =\frac{XX'}{2 \sum_{j=1}^{nSNP} p_{j} (1 - p_{j})}
\end{equation}

]

--

<br>

<P align="justify">La heredabilidad en sentido estricto se estimó mediante el método BLUP como:</p>

\begin{equation}
h^{2} = \frac{\sigma^{2}_{g}}{\sigma^{2}_{g} + \sigma^{2}_{e}}
\end{equation}

---
class: center

### Materiales y métodos (parciales)

<P align="justify"><b>Precisión de la predición mediante simulación ancestral</b></p>

.pull-left[

<P align="justify">Se utilizó la metodología del software Molcoanc<sup>1</sup> con el fin de construir tres pedigríes.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'Pedigrí', ~'Núm. de individuos',
  'Pedigrí 1', '300 + 451 = 751',
  'Pedigrí 2', '1210 + 451 = 1661',
  'Pedigrí 3', '2000 + 451 = 2451'
  ) %>%
  gt()
```

<P align="justify">Este proceso se replicó diez veces para cada pedigrí.</p>

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE, message = FALSE}

Puntos <- data.frame(x = c(1.0, 1.5, 0.75, 1.25, 1.75, 0.5, 1.0, 1.5, 2.0, 0.25, 0.75, 1.25, 1.75, 2.25), y = c(c(2.8, 2.8, 1.5, 1.5, 1.5, 0.2, 0.2, 0.2, 0.2, -1.1, -1.1, -1.1, -1.1, -1.1)))

data.frame(
  x = 1.0, y = 1.0
  ) %>%
  ggplot() +
  
  annotate(geom = 'rect', xmin = -0.4, xmax = 2.5, ymin = -1.7, ymax = 3.4, colour = 'black', fill = 'white', size = 0.4, linetype = 'dashed') +
  geom_point(aes(x = x, y = y), data = Puntos, size = 5.6, shape = 21, color = 'black', fill = 'black', alpha = 0.8) +
  geom_curve(x = 0.05, xend = 0.05, y = -1.1, yend = 2.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  annotate(geom = 'text', x = -0.3, y = 0.85, label = 'Creación de antepasados de indivi-', size = 5.2, colour = 'black', angle = 90) +
  annotate(geom = 'text', x = -0.1, y = 0.85, label = 'duos genotipados mediante simulación', size = 5.2, colour = 'black', angle = 90) +
  
  annotate(geom = 'segment', x = 1.0, xend = 0.75, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.5, xend = 0.75, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.0, xend = 1.25, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.5, xend = 1.25, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.5, xend = 1.75, y = 2.8, yend = 1.5, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 0.75, xend = 0.5, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.25, xend = 1.0, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.25, xend = 2.0, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.75, xend = 2.0, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.75, xend = 1.5, y = 1.5, yend = 0.2, colour = 'black', size = 0.8) +
  
  annotate(geom = 'segment', x = 0.25, xend = 0.5, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 0.75, xend = 1.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.25, xend = 1.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 1.75, xend = 2.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  annotate(geom = 'segment', x = 2.25, xend = 2.0, y = -1.1, yend = 0.2, colour = 'black', size = 0.8) +
  
  geom_curve(x = 2.6, xend = 3.0, y = 2.8, yend = 2.8, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 3.1, xmax = 4.1, ymin = 3.0, ymax = 1.8, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 3.6, y = 2.4, label = 'A', size = 10.4, colour = 'black') +
  
  scale_x_continuous(limits = c(-0.4, 4.2)) +
  scale_y_continuous(limits = c(-1.8, 3.4)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:24")`.</FONT></p>

]

---
class: center

### Materiales y métodos (parciales)

<P align="justify"><b>Precisión de la predición mediante simulación ancestral</b></p>

.pull-left[

<P align="justify">Se usaron diferentes subconjuntos de datos con la siguientes características:</p>

<P align="justify">a) Distinta cantidad de individuos genotipados.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'148', ~'298 ', ~'451'
  ) %>%
  gt()
```

<P align="justify">b) Diferentes densidades de marcador.</p>

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', message = FALSE}

tribble(
  ~'1.000', ~'10.000 ', ~'100.000'
  ) %>%
  gt()
```

]

.pull-right[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE, warning = FALSE, message = FALSE}

data.frame(
  x = 1.0, y = 1.0
  ) %>%
  ggplot() +
  
  annotate(geom = 'text', x = 0.5, y = 3.5, label = 'Ind. genotipados', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 1.5, y = 3.5, label = '148', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 2.5, y = 3.5, label = '298', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 3.5, y = 3.5, label = '298', size = 6.2, colour = 'black') +
  
  annotate(geom = 'text', x = 0.5, y = 2.0, label = 'Densidad de SNP', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 1.5, y = 2.0, label = '1.000', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 2.5, y = 2.0, label = '10.000', size = 6.2, colour = 'black') +
  annotate(geom = 'text', x = 3.5, y = 2.0, label = '100.000', size = 6.2, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 1.1, xmax = 1.9, ymin = 1.0, ymax = 0.4, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 1.5, y = 0.7, label = 'G', size = 9.4, colour = 'black') +
  annotate(geom = 'rect', xmin = 2.1, xmax = 2.9, ymin = 1.0, ymax = 0.4, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 2.5, y = 0.7, label = 'G', size = 9.4, colour = 'black') +
  annotate(geom = 'rect', xmin = 3.1, xmax = 3.9, ymin = 1.0, ymax = 0.4, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 3.5, y = 0.7, label = 'G', size = 9.4, colour = 'black') +
  
  geom_curve(x = 1.5, xend = 1.5, y = 3.3, yend = 2.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 1.5, xend = 2.5, y = 3.3, yend = 2.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 1.5, xend = 3.5, y = 3.3, yend = 2.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  
  geom_curve(x = 1.5, xend = 1.5, y = 1.8, yend = 1.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 2.5, xend = 2.5, y = 1.8, yend = 1.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  geom_curve(x = 3.5, xend = 3.5, y = 1.8, yend = 1.2, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0) +
  
  scale_x_continuous(limits = c(0.0, 3.9)) +
  scale_y_continuous(limits = c(0.4, 3.6)) +
  
  theme_bw() +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank())
```

]

---
class: center

### Materiales y métodos (parciales)

<P align="justify"><b>Precisión de la predición mediante simulación ancestral</b></p>

.pull-left[

```{r, echo = FALSE, out.width = "100%", fig.align = 'center', fig.showtext = TRUE}

data.frame(
  x = 4.5, y = 0.5
  ) %>%
  ggplot(aes(x0 = x, y0 = y, r = 0.8)) +
  #geom_blank() +
  
  annotate(geom = 'rect', xmin = 1.5, xmax = 4.25, ymin = 7.5, ymax = 9.5, colour = 'yellow', fill = 'yellow', alpha = 0.2) +
  annotate(geom = 'text', x = 2.875, y = 8.5, label = 'A', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 1.1, y = 8.5, label = 'Matriz A', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 7.0, xmax = 8.75, ymin = 7.75, ymax = 9.0, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 7.875, y = 8.375, label = 'G', size = 8.4, colour = 'black') +
  annotate(geom = 'text', x = 6.6, y = 8.375, label = 'Matriz G', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 6.0, xmax = 8.75, ymin = 3.75, ymax = 5.75, colour = 'black', fill = 'white', alpha = 0.2) +
  annotate(geom = 'rect', xmin = 6.8, xmax = 8.55, ymin = 3.95, ymax = 5.05, colour = 'cyan', fill = 'cyan', alpha = 0.2) +
  annotate(geom = 'text', x = 5.6, y = 4.75, label = 'Matriz H', size = 7.6, angle = 90, colour = 'black') +
  
  annotate(geom = 'rect', xmin = 6.275, xmax = 8.475, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 7.375, y = 1.8, label = 'ssGBLUP', size = 7.6, colour = 'black') +
  geom_curve(x = 7.375, xend = 7.375, y = 3.45, yend = 2.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  annotate(geom = 'rect', xmin = 1.55, xmax = 3.75, ymin = 1.4, ymax = 2.2, colour = 'black', fill = 'white', alpha = 0.2, linetype = 'dashed') +
  annotate(geom = 'text', x = 2.65, y = 1.8, label = 'BLUP', size = 7.6, colour = 'black') +
  
  geom_curve(x = 3.0, xend = 5.2, y = 7.3, yend = 4.9, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  geom_curve(x = 7.875, xend = 7.875, y = 7.45, yend = 6.05, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'solid') +
  geom_curve(x = 2.65, xend = 2.65, y = 7.3, yend = 2.5, arrow = arrow(length = unit(0.1, 'inch')), size = 1.2, color = 'black', curvature = 0.0, linetype = 'dashed') +
  
  scale_x_continuous(limits = c(0.45, 9.4)) +
  scale_y_continuous(limits = c(1.0, 9.9)) +
  
  theme_bw() +
  #labs(title = 'ssGBLUP') +
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        axis.title = element_blank(),
        plot.title = element_text(size = 11, face = 'bold'))
```

]

--

.pull-right[

<P align="justify">Se usaron los paquetes `BGLR`<sup>1</sup> y `lme4GS`<sup>2</sup> del `R` para estimar los componentes de varianza y los valores fenotípicos predichos, permitiendo así la predicción mediante procedimientos Bayesianos y REML.</p>

<br>

<P align="justify">Se uso el coeficiente de correlación entre los valores fenotípicos observados y predichos como medida de predictibilidad (48 individuos mejorados).</p>

]

<br>

.footnote[

<P align="justify"><FONT SIZE=2>[1] `r Citet(mi_Bib, "cite:50")`.</FONT></p>
<P align="justify"><FONT SIZE=2>[2] `r Citet(mi_Bib, "cite:51")`.</FONT></p>

]

---
class: center

### Resultados



---

### Bibliografía

```{r refs 1, echo=FALSE, results="asis"}

PrintBibliography(bib = mi_Bib, start = 1, end = 3)
```

---

class: center, middle, inverse

<span class = "cc">
<!-- Key: a: SA, b: BY, c: CC Circle, d: ND, n: NC, m: Sampling, s: Share, r: Remix, C: CC Full Logo -->
C bn
</span>

[Licencia CC BY-NC](https://creativecommons.org/licenses/by-nc/4.0/)
